examples/aligned_accessor/aligned_accessor.cpp

```diff
@@ -0,0 +1,714 @@
+		stdex::default_accessor<ElementType>>;
+
+#define TICK() const auto tick = std::chrono::steady_clock::now()
+
+using double_seconds = std::chrono::duration<double, std::ratio<1>>;
+
+#define TOCK() std::chrono::duration_cast<double_seconds>(std::chrono::steady_clock::now() - tick).count()
+
+template<class ElementType, std::size_t byte_alignment>
+void add_aligned_mdspan_1d(aligned_mdspan_1d<const ElementType, byte_alignment> x,
+			   aligned_mdspan_1d<const ElementType, byte_alignment> y,
+			   aligned_mdspan_1d<ElementType, byte_alignment> z)
+{
+  const index_type n = z.extent(0);
+  for (index_type i = 0; i < n; ++i) {
+    z[i] = x[i] + y[i];
+  }
+}
+
+template<class ElementType, std::size_t byte_alignment>
+auto benchmark_add_aligned_mdspan_1d(const std::size_t num_trials,
+				     const index_type n,
+				     const ElementType x[],
+				     const ElementType y[],
+				     ElementType z[],
+				     std::integral_constant<std::size_t, byte_alignment> /* ba */ )
+{
+  TICK();
+  aligned_mdspan_1d<const ElementType, byte_alignment> x2{x, n};
+  aligned_mdspan_1d<const ElementType, byte_alignment> y2{y, n};
+  aligned_mdspan_1d<ElementType, byte_alignment> z2{z, n};
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_aligned_mdspan_1d(x2, y2, z2);
+  }
+  return TOCK();
+}
+
+template<class ElementType>
+void add_mdspan_1d(mdspan_1d<const ElementType> x,
+		   mdspan_1d<const ElementType> y,
+		   mdspan_1d<ElementType> z)
+{
+  const index_type n = z.extent(0);
+  for (index_type i = 0; i < n; ++i) {
+    z[i] = x[i] + y[i];
+  }
+}
+
+template<class ElementType>
+auto benchmark_add_mdspan_1d(const std::size_t num_trials,
+			     const index_type n,
+			     const ElementType x[],
+			     const ElementType y[],
+			     ElementType z[])
+{
+  TICK();
+  mdspan_1d<const ElementType> x2{x, n};
+  mdspan_1d<const ElementType> y2{y, n};
+  mdspan_1d<ElementType> z2{z, n};
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_mdspan_1d(x2, y2, z2);
+  }
+  return TOCK();
+}
+
+template<class ElementType>
+void add_raw_1d(const index_type n,
+		const ElementType x[],
+		const ElementType y[],
+		ElementType z[])
+{
+  for (index_type i = 0; i < n; ++i) {
+    z[i] = x[i] + y[i];
+  }
+}
+
+template<class ElementType>
+auto benchmark_add_raw_1d(const std::size_t num_trials,
+			  const index_type n,
+			  const ElementType x[],
+			  const ElementType y[],
+			  ElementType z[])
+{
+  TICK();
+  for (std::size_t trial = 0; trial < num_trials; ++trial) {
+    add_raw_1d(n, x, y, z);
+  }
+  return TOCK();
+}
+
+// Assume that x, y, and z all have the same alignment.
+template<class ElementType, std::size_t byte_alignment>
+void add_aligned_raw_1d(const index_type n,
+			aligned_pointer_t<const ElementType, byte_alignment> x,
+			aligned_pointer_t<const ElementType, byte_alignment> y,
+			aligned_pointer_t<ElementType, byte_alignment> z,
+			std::integral_constant<std::size_t, byte_alignment> /* ba */ )
```

Comment:
Why do you bother with that argument

Reply:
1. Consistency with the aligned mdspan interface
2. `aligned_pointer_t<ElementType, byte_alignment` is an alias and might just be `ElementType*`, but it might be a pointer with an attribute.  I wasn't sure whether compilers could deduce `byte_alignment` from an attribute on the pointer.

I'll experiment with removing this parameter.

Reply:
I was able to remove this parameter; thanks!